iroha_p2p/tests/mod.rs

```diff
@@ -0,0 +1,156 @@
+    info!("Starting network tests...");
+    let broker = Broker::new();
+    let network = Network::<TestMessage>::new(broker.clone(), String::from("127.0.0.1:4433"))
+        .await
+        .unwrap();
+    drop(network.start().await);
+    tokio::time::sleep(Duration::from_millis(200)).await;
+
+    info!("Connecting to peer...");
+    let peer1 = PeerId {
+        address: String::from("127.0.0.1:4433"),
+        public_key: None,
+    };
+    broker.issue_send(Connect { id: peer1.clone() }).await;
+    tokio::time::sleep(Duration::from_millis(200)).await;
+
+    info!("Posting message...");
+    broker
+        .issue_send(Post {
+            data: TestMessage("Some data to send to peer".to_owned()),
+            id: peer1,
+        })
+        .await;
+
+    tokio::time::sleep(Duration::from_millis(200)).await;
+}
+
+#[derive(Debug)]
+pub struct TestActor {
+    broker: Broker,
+    messages: Arc<AtomicU32>,
+}
+
+#[async_trait::async_trait]
+impl Actor for TestActor {
+    async fn on_start(&mut self, ctx: &mut Context<Self>) {
+        self.broker.subscribe::<TestMessage, _>(ctx);
+    }
+}
+
+#[async_trait::async_trait]
+impl Handler<TestMessage> for TestActor {
+    type Result = ();
+
+    async fn handle(&mut self, msg: TestMessage) -> Self::Result {
+        info!("Actor got message {:?}", msg);
+        let _ = self.messages.fetch_add(1, Ordering::Relaxed);
+    }
+}
+
+/// This test creates two networks and one peer from the first network.
+/// This peer connects to our second network, emulating some distant peer.
+#[tokio::test(flavor = "multi_thread", worker_threads = 4)]
+async fn two_networks() {
+    let log_config = LoggerConfiguration {
+        max_log_level: LevelEnv::TRACE,
+        compact_mode: false,
+        ..LoggerConfiguration::default()
+    };
+    drop(iroha_logger::init(log_config));
+    info!("Starting first network...");
+    let broker = Broker::new();
+    let network1 = Network::<TestMessage>::new(broker.clone(), String::from("127.0.0.1:4434"))
+        .await
+        .unwrap();
+    drop(network1.start().await);
+    tokio::time::sleep(Duration::from_millis(200)).await;
+
+    info!("Starting second network...");
+    let broker2 = Broker::new();
+    let network2 = Network::<TestMessage>::new(broker2.clone(), String::from("127.0.0.1:4435"))
+        .await
+        .unwrap();
+    drop(network2.start().await);
+    tokio::time::sleep(Duration::from_millis(200)).await;
+
+    let messages = Arc::new(AtomicU32::new(0));
+    let actor = TestActor {
+        broker: broker2.clone(),
+        messages: Arc::clone(&messages),
+    };
+    drop(actor.start().await);
+    tokio::time::sleep(Duration::from_millis(200)).await;
+
+    info!("Connecting to peer...");
+    let peer1 = PeerId {
+        address: String::from("127.0.0.1:4435"),
+        public_key: None,
+    };
+    broker.issue_send(Connect { id: peer1.clone() }).await;
+    tokio::time::sleep(Duration::from_millis(200)).await;
```

Comment:
Please keep the indexes logically consistent e.g broker1, network1, peer1 should all correspond to port 4434, and the ones with 2 to port 4435. Or you can name them broker_4434, network_4434 etc. alternatively.

Reply:
I guess it can just use `unique_port` crate. It would be way better

Reply:
My main comment here was about consistency of variable names, that they shouldn't be mixed between logically different (that would be in real life on different machines) nodes.

Reply:
But unique_port can also be used, then it should be broker1, network1, peer1 relates to one IP as I previously suggested

Reply:
Yes, that is what i meant. Generally dont like some random constants in variable names. It doesnt add context and kinda confusing

Reply:
Changed to `unique_port`.